synapse/handlers/sliding_sync.py

```diff
@@ -1403,36 +1455,22 @@ async def get_room_sync_data(
             prev_batch_token = to_token
 
             # We're going to paginate backwards from the `to_token`
-            from_bound = to_token.room_key
+            to_bound = to_token.room_key
             # People shouldn't see past their leave/ban event
             if room_membership_for_user_at_to_token.membership in (
                 Membership.LEAVE,
                 Membership.BAN,
             ):
-                from_bound = (
+                to_bound = (
                     room_membership_for_user_at_to_token.event_pos.to_room_stream_token()
                 )
 
-            # Determine whether we should limit the timeline to the token range.
-            #
-            # We should return historical messages (before token range) in the
-            # following cases because we want clients to be able to show a basic
-            # screen of information:
-            #  - Initial sync (because no `from_token` to limit us anyway)
-            #  - When users `newly_joined`
-            #  - TODO: For an incremental sync where we haven't sent it down this
-            #    connection before
-            to_bound = (
-                from_token.stream.room_key
-                if from_token is not None
-                and not room_membership_for_user_at_to_token.newly_joined
-                else None
-            )
-
             timeline_events, new_room_key = await self.store.paginate_room_events(
                 room_id=room_id,
-                from_key=from_bound,
-                to_key=to_bound,
+                # Because we want the latest events first the bounds are
+                # reversed.
```

Comment:
This comment reads a bit ambiguous to me. We do want to get the "latest events first" in terms of filling up the `limit`. When I read this, I also get confused about it trying to describe the order of events we want. Especially, since we actually want the events in ascending order in the end so we `timeline_events.reverse()` below (oldest to newest events).

I think it's more that we're paginating backwards that needs to be called out:

```suggestion
                # The bounds are reversed so we can paginate backwards
                # (from newer to older events) starting at to_bound.
                # This ensures we fill the `limit` with the newest events first,
```

Reply:
yeah makes sense